ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/common/ActionFactory.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
```

Comment:
This is a factory for common actions, at this moment for collapse/expand tree. It can be used to create dynamic actions that should not be registered in the `ActionManager` but displayed on the any tree you want.